### PR TITLE
feat: Add precedence evaluator for sidecar injection decisions

### PR DESCRIPTION
**Pull request by cwiklik**
## Summary
- Add `PrecedenceEvaluator` that determines sidecar injection via 6-layer precedence chain: global gate → per-sidecar gate → namespace label → workload label → TokenExchange CR → platform defaults
- Add `InjectionDecision`, `SidecarDecision`, and `TokenExchangeOverrides` types
- Add label constants for per-sidecar workload opt-out
- proxy-init always follows envoy-proxy decision
- 26 table-driven tests covering all precedence layers and edge cases

Part of #109 (Phase 2 of 5)
Depends on: #110 (Phase 1 — config types & loaders)

## Test plan
- [x] `go test ./internal/webhook/injector/...` — 26 tests pass
- [x] `go test -race ./internal/webhook/injector/...` — no data races
- [x] `go vet ./...` — clean
- [x] `go build -race ./...` — clean
- [x] `golangci-lint run` — 0 findings on new files
- [x] `staticcheck` — clean
- [x] `govulncheck` — 0 vulnerabilities in code

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: kagenti-webhook/internal/webhook/injector/precedence.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	}
+
+	// Layer 6: Platform defaults
+	if !platformDefaultEnabled {
+		return SidecarDecision{
+			Inject: false,
+			Reason: "platform default disabled " + sidecarName,
+			Layer:  "platform-default",
+		}
+	}
```
[the remark this answers is not in the document]
Comment (cwiklik) [reply kagenti-webhook/internal/webhook/injector/precedence.go]: Why this is intentional: The precedence chain is designed as veto-only (short-circuit "no" only), not authority-based opt-in/opt-out:

Each layer can veto (force Inject=false) but cannot override a higher layer's veto
This ensures operators maintain control via feature gates and platform defaults regardless of workload requests
Test case "CRD enables sidecar that higher layer disabled - still skipped" validates this behavior
Changing this would be an architectural change requiring epic-level discussion, not a Phase 2 bug fix
No fix needed - this is the intended design.
